Remove the concept of a Task

WilliamMcCumstie opened the pull request:
Tasks are now indistinguishable from other jobs

WilliamMcCumstie committed:
Remove the concept of a Task

Tasks are now indistinguishable from other jobs

benarmston commented:
This looks a lot simpler.  There is an issue with submitting a single array task:

```
$ ./bin/scheduler batch --array 1 --nodes 1 tmp/simple-array.sh  this argument
Submitted batch job 0b9f5d23-33b9-46e0-ade4-1c2ae966e3fc[-1]
$ ./bin/scheduler queue
┌─────────────────────────────────────────┬───────────┬────────┬──────┬─────────┬──────┬───────┬──────────────────┐
│ JOBID                                   │ PARTITION │ NAME   │ USER │ ST      │ TIME │ NODES │ NODELIST(REASON) │
├─────────────────────────────────────────┼───────────┼────────┼──────┼─────────┼──────┼───────┼──────────────────┤
│ 0b9f5d23-33b9-46e0-ade4-1c2ae966e3fc[1] │ all       │ (none) │ TBD  │ RUNNING │ TBD  │ 1     │ node01           │
└─────────────────────────────────────────┴───────────┴────────┴──────┴─────────┴──────┴───────┴──────────────────┘
```

The job id in the output of `batch` doesn't look quite right.  As best I can tell this is related to determining the ID to return after the job has been scheduled and allocated resources.  (So this probably ought to be on the controller PR, but I started typing it against this PR).

benarmston commented:
The above issue has been fixed by commits on the controller repo.

WilliamMcCumstie committed:
Add a compatibility fix when outputting ARRAY_JOB id

In this case the `ID` should not include the array indexing,
however it should include it in the `queue` output

However as they are the same resource, we can't have it both
ways
